pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -469,6 +498,58 @@ def final_transform(self) -> bool:
         """``True`` if the transform needs to be executed"""
         return self._transform_dispatcher.final_transform
 
+    def __add__(self, other):
+        """Add two containers or a container and a dispatcher to create a TransformProgram.
+
+        For container + program, Python falls back to TransformProgram.__radd__.
+
+        Args:
+            other: Another TransformContainer or TransformDispatcher to add.
+
+        Returns:
+            TransformProgram: A new program with this container followed by the other.
+        """
+        # Convert dispatcher to container if needed
+        if isinstance(other, TransformDispatcher):
+            other = TransformContainer(other)
+
+        if isinstance(other, TransformContainer):
+            # Import here to avoid circular import
+            from .transform_program import TransformProgram  # pylint: disable=import-outside-toplevel
+
+            if self.final_transform and other.final_transform:
+                raise TransformError(
+                    f"Both {self} and {other} are final transforms and cannot be combined."
+                )
+            return TransformProgram([self, other])
+
+        # For TransformProgram, Python falls back to program.__radd__(container)
+        return NotImplemented
+
+    def __mul__(self, n):
+        """Multiply a container by an integer to create a program with repeated containers.
+
+        Args:
+            n (int): Number of times to repeat this container.
+
+        Returns:
+            TransformProgram: A new program with this container repeated n times.
+        """
+        # Import here to avoid circular import
+        from .transform_program import TransformProgram  # pylint: disable=import-outside-toplevel
+
+        if isinstance(n, int):
+            if n < 0:
+                raise ValueError("Cannot multiply transform container by negative integer")
+            if self.final_transform and n > 1:
+                raise TransformError(
+                    f"{self} is a final transform and cannot be applied more than once."
+                )
+            return TransformProgram([self] * n)
+        return NotImplemented
```

Comment:
Just reducing the level of nesting.
```suggestion
        if not isinstance(n, int):
            return NotImplemented

        if n < 0:
            raise ValueError("Cannot multiply transform container by negative integer")

        if self.final_transform and n > 1:
            raise TransformError(
                f"{self} is a final transform and cannot be applied more than once."
            )

        return TransformProgram([self] * n)
```